### PR TITLE
Corrected unit test

### PR DESCRIPTION
**Pull request by prasadsom**
Fixed Factor files unit test
#### Description 
There are two issues here. After downloading data from yahoo. Split factor is assigned as whole number instead of fraction. ie. 7 instead of 1/7
Second issue, in my opinion aapl.csv price factor numbers are slightly different from what they should be.
That's my assumption based on calculation of price factor in code. 
https://github.com/QuantConnect/Lean/blob/dfbbf0aba76a8023f83f84a2cbfd770879f2bd62/ToolBox/FactorFileGenerator.cs#L229

for date 20180208 according to calculation, which I think is correct ,
```
previousFactorFileRow.PriceFactor = 0.9961585
dividend.Value = 0.63
previousClosingPrice.Close = 155.16
previousFactorFileRow.SplitFactor = 1

```
with these values `priceFactor = 0.9920982` . Over 29 rows, difference between what's in file and calculated value falls outside tolerance. And hence test will fail. That's why I modified aapl.csv file
#### Related Issue

#3435 
#### Requires Documentation Change

NA
#### How Has This Been Tested?

Passed unit test
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: ToolBox/YahooDownloader/YahooDataDownloader.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public Queue<BaseData> DownloadSplitAndDividendData(Symbol symbol, DateTime star
                         parsed.Add(new Split
                         {
                             Time = Parse.DateTimeExact(values[0].Replace("-", string.Empty), DateFormat.EightCharacter),
-                            Value = ParseAmount(values[1])
+                            Value = 1/ParseAmount(values[1])
```
Comment (prasadsom) [inline ToolBox/YahooDownloader/YahooDataDownloader.cs]: In present form it's assigning whole value. eg. for 1 to 7 split. It's assigning 7 instead of 1/7

**Comment by prasadsom** [comment]
> A possible solution is copying the test factor files into the TestData folder and redirect the test to that file.
In that , case would it be better to do comparison in different way in unit test?  May be by increasing tolerance for price factor comparison. In this case, two places after decimal(0.01) ?

**Comment by Jay-Jay-D** [comment]
One of the characteristics of a good unit test is isolation, it shouldn't have a dependency on any outside factor.

If you check `AAPL` factor file history, you can see it changes. Therefore, if the sample data changes or is updated, it'll break the test.

That's why I insist on adding the testing factor file into the test data folder.

**Comment by prasadsom** [comment]
Thanks @Jay-Jay-D , @Martin-Molinero I changed location of inverting split value. And modified calculation of pricefactor as mentioned in PR #3567 . This will fix unit test without changing factor file

**Pull request merged**